bpf/bpf_overlay.c

```diff
@@ -674,9 +674,13 @@ int cil_from_overlay(struct __ctx_buff *ctx)
 			struct bpf_tunnel_key key = {};
 			__u32 key_size = TUNNEL_KEY_WITHOUT_SRC_IP;
 
-			if (unlikely(ctx_get_tunnel_key(ctx, &key, key_size, 0) < 0)) {
-				ret = DROP_NO_TUNNEL_KEY;
-				goto out;
+			ret = ctx_get_tunnel_key(ctx, &key, key_size, 0);
+			if (ret < 0) {
+				ret = ctx_get_tunnel_key(ctx, &key, key_size, BPF_F_TUNINFO_IPV6);
+				if (unlikely(ret < 0)) {
+					ret = DROP_NO_TUNNEL_KEY;
+					goto out;
+				}
```

Comment:
oh yuck :)

Reply:
Yeah... I could probably hide that stuff behind a helper function though :thinking: